Sources/FaunaDB/JSON.swift

```diff
@@ -197,6 +197,26 @@ private extension JsonType {
         }
     }
 
+    private func toRefV(object: [String: JsonType]) throws -> Value {
+        guard case let .string(id)? = object["id"] else { throw JsonError.invalidReference }
+
+        if object["class"] == nil && object["database"] == nil {
+            return Native.fromName(id)
+        }
+
+        var clazz: RefV? = nil
+        if case let .object(cls)? = object["class"] {
+            clazz = try toValue(special: cls) as? RefV
```

Comment:
I think there is an issue with this `as?`. If the value returned is not a `RefV`, it will become `nil` instead of throwing an error. That will allow for wrong formatted ref components to be ignored, which might not be what we want.

Reply:
Ok, I had to change completely the logic here, I figured out that in the first `if` if something different than an `object` was parsed it will not even enter in the condition. Also I kept using `as?` because the linter does not allow `as!` so I checked to see if the conversion was something different than nil and throw an exception otherwise.